src/shared/lp-faas-toolbelt/context-service-sdk/util/helper.ts

```diff
@@ -0,0 +1,27 @@
+import { ErrorCodes } from './constants';
+import { SDKError } from '../types/errors/sdkError';
+
+let timerID: NodeJS.Timeout; // Required in order to allow cleanup of timeouts
```

Comment:
This needs to be moved inside `runTaskWithDeadline` otherwise there will be a leak on parallel calls or other issues with parallel calls.

```suggestion
let timerID: NodeJS.Timeout; // Required in order to allow cleanup of timeouts
```

Reply:
You're right. The compiler complained about `Variable 'timerID' is used before being assigned.`. 
Have to to initialize `timerID` with undefined.
```suggestion
let timerID: (NodeJS.Timeout | undefined) = undefined; // Required in order to allow cleanup of timeouts
```